### PR TITLE
Extract OCR via a background job

### PR DESCRIPTION
**Pull request by tpendragon**

### CODE REVIEW

File: app/models/book.rb [hunk 1]
```diff
@@ -27,15 +27,7 @@ def manifests
   end
 
   def range_labels_from_manifests
```
Comment (hackmastera) [inline app/models/book.rb]: can you add a comment as to how the ocr text is related to the range labels from manifests?

Comment (hackmastera) [reply app/models/book.rb]: or change the method name?

File: app/jobs/populate_version_ocr_job.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class PopulateVersionOCRJob < ApplicationJob
```
Comment (hackmastera) [inline app/jobs/populate_version_ocr_job.rb]: where is this new job tested?

File: app/models/version.rb [hunk 3]
```diff
@@ -1,17 +1,23 @@
-class Version < ActiveRecord::Base
+class Version < ApplicationRecord
   belongs_to :contributing_library
   belongs_to :book
   validates :label, :contributing_library, :owner_system_number, presence: true
   validates :manifest, :rights, url: true
   validates :based_on_original, inclusion: { in: [true, false] }
 
   after_commit :update_index
+  after_commit :populate_ocr_text, if: :saved_change_to_manifest?, on: [:create, :update]
```
Comment (hackmastera) [inline app/models/version.rb]: test that the job is queued?

File: app/models/book.rb (hunk 1 above)
Comment (escowles) [reply app/models/book.rb]: I think the job needs to be renamed to IndexRangeLabelsJob or something, because it's indexing range labels, not OCR.

Comment (tpendragon) [reply app/models/book.rb]: It's -going- to be OCR at one point, and range labels are kinda like that.

Comment (escowles) [reply app/models/book.rb]: OK, that makes sense. Can you add a comment or something to clarify the disconnect, so it's clear what's happening now vs. what we plan on doing?

**Comment by tpendragon** [comment]
@hackmastera I implemented the features. I'll have to wait until https://github.com/pulibrary/princeton_ansible/pull/456 is merged to add a PR to add sidekiq to staging.

**Comment by hackmastera** [comment]
@tpendragon what did you decide about a test for the job itself?

**Comment by tpendragon** [comment]
> @tpendragon what did you decide about a test for the job itself?

Oh, sorry. Yeah, it's integration tested right now. I'll add a test to the job and remove the integration.

**Comment by tpendragon** [comment]
@hackmastera Confirmed this works in staging. The ansible PR is here: https://github.com/pulibrary/princeton_ansible/pull/457

**Pull request merged**